File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.c
#include <stdio.h>
#include <stdbool.h>

bool solution(int x) {
    bool answer = true;
    int sum = 0, temp = x;
while(temp / 10 > 0){
    sum += temp%10;
    temp = (temp-temp%10)/10;
}
    sum += temp%10;
    if(x % sum != 0)
        answer = false;
    return answer;
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.py
def solution(x):
    answer = True
    temp = list(map(int,str(x))) # map(int,)를 안사용하면 문자열을 sum 하려는거라서 type에러 발생
    if x % sum(temp) != 0:
        answer = False
    return answer